.github/workflows/gradle.yml

```diff
@@ -17,20 +17,93 @@ on:
 jobs:
   build:
 
-    runs-on: macOS-latest
-
     strategy:
       matrix:
-        java: [ 8, 11, 17 ]
+        include:
+          - java: 8
+            platform: macos-latest
+            e2e-tests: android
+          - java: 11
+            platform: macos-latest
+            e2e-tests: ios
+          - java: 17
+            platform: ubuntu-latest
```

Comment:
what do we actually do for this step?

Reply:
Build client on Java 17 to be sure it's compatible (actually this step was here before, now it's another form of configuration)

Reply:
it would probably make more sense to dynamically select recent java LTS versions and run build for all of them in parallel. See https://github.com/appium/appium-xcuitest-driver/blob/22814ea6f849b69484e5f0becd13f1f90b209dd2/.github/workflows/unit-test.yml#L6 for more examples on the dynamic matrix generation

Reply:
it can be an overkill for now, there are 3 LTS Java-s: 8, 11, 17. 8 and 11 are declared explicitly as E2E tests are run on them. Only 17 is left.
Java 21 (LTS) is going to be released on September. But at the same time Selenium team is talking about bumping Java to 11. So I see a lot of uncertainty here

Reply:
could be. Lets just put a TODO there to add the new LTS as soon as it is released